### PR TITLE
Add c-ares port

### PR DESCRIPTION
**Pull request by donny-dont**
Adds c-ares v1.16.1. Updates the curl port with an option to enable it as the DNS resolver.

### CODE REVIEW

File: WindowsRequirements.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
     "brotli",
     "libressl",
     "nghttp2",
-    "curl[ca-bundle,ssl,ipv6]",
+    "c-ares",
+    "curl[ares,ca-bundle,ssl,ipv6]",
```
Comment (foopoiuyt) [inline WindowsRequirements.json:7]: I think we should probably not yet default curl to using ares until the network guys are a bit more ready

**Review by foopoiuyt — CHANGES_REQUESTED** [review]
One small thing to change, but otherwise seems good.

File: WindowsRequirements.json (hunk 1 above)
Comment (donny-dont) [reply WindowsRequirements.json]: Fixed

**Review by foopoiuyt — APPROVED** [review]
LGTM

**Pull request merged**